File: app/src/main/java/com/katyabaygin/menu/DrinkDao.java
package com.katyabaygin.menu;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface DrinkDao {
    @Insert
    void insert(Drink drink);

    @Delete
    void delete(Drink drink);

    @Update
    void updateDrink(Drink drink);


    @Query("SELECT * FROM drinks WHERE idDrink = :id")
    Drink getDrinkById(int id);


    @Query("SELECT * FROM drinks")
    LiveData<List<Drink>> getAllDrinksLiveData();


}

File: app/src/main/java/com/katyabaygin/menu/data/DrinkDao.kt
package com.katyabaygin.menu.data

import androidx.lifecycle.LiveData
import androidx.room.*
import com.katyabaygin.menu.domain.Drink

@Dao
interface DrinkDao {
    @Insert
    fun insert(drink: Drink)

    @Delete
    fun delete(drink: Drink)

    @Update
    fun updateDrink(drink: Drink)

    @Query("SELECT * FROM drinks WHERE idDrink = :id")
    fun getDrinkById(id: String): Drink

    @Query("SELECT * FROM drinks")
    fun getAllDrinksLiveData(): LiveData<List<Drink>>
}
